estate/models/estate_property_offer.py

```diff
@@ -44,10 +49,35 @@ def action_confirm(self):
         accepted_offer = self.property_id.offer_ids.filtered(lambda o : o.status == 'accepted')
         if accepted_offer:
             raise exceptions.UserError("More than one offer cannot be accepted")
+        if self.property_id.expected_price and self.price < self.property_id.expected_price * 0.9:
+            raise exceptions.UserError("The selling price must be at least 90% of the expected price! You must reduce the expected price if you want to accept this offer'")
         self.status = "accepted"
         self.property_id.selling_price = self.price
         self.property_id.buyer_id = self.buyer_id
 
     def action_deny(self):
         for record in self:
             record.status = "refused"
+
+    _sql_constraints = [
+        (
+            "check_price",
+            "CHECK(price > 0)",
+            "Offer price must be strictly positive",
+        )
+    ]
+
+    @api.model
+    def create(self, vals):
+        property_obj = self.env['property'].browse(vals['property_id'])
+        property_obj.state = 'offer_received'
+        offers = property_obj.offer_ids
+        max_offer = max(offers.mapped("price"), default=0)
+        if vals["price"] < max_offer:
+            raise exceptions.UserError("Cannot create an offer with a lower amount than an existing offer.")
+        return super(PropertyOffer, self).create(vals)
```

Comment:
class and self is not needed for python3
```suggestion
        return super().create(vals)
```